lib/ruby_lsp/requests/diagnostics.rb

```diff
@@ -19,6 +19,10 @@ module Requests
     class Diagnostics < Request
       extend T::Sig
 
+      # Computing "external diagnostics", like RuboCop violations, can be very expensive in super large files. Similar
+      # to semantic highlighting, we limit the feature in favor of performance and responsiveness
+      MAXIMUM_CHARACTERS_FOR_EXTERNAL_DIAGNOSTICS = 100_000
```

Comment:
Should we just use `MAXIMUM_CHARACTERS_FOR_HIGHLIGHT` and maybe rename it to `MAXIMUM_CHARACTERS_FOR_HIGHLIGHT_AND_DIAGNOSTICS` (or something more generic)? Otherwise, there could be a mismatch between features and messages' condition over time. Like if `MAXIMUM_CHARACTERS_FOR_HIGHLIGHT` is reduced, the message in did open request will disappear in some cases while the diagnostics is still disabled.